### PR TITLE
Add "Développement d'applications avec Objective Caml"

### PR DESCRIPTION
**Pull request by cwfoo**
## What does this PR do?
Add book: [Développement d'applications avec Objective Caml](https://www-apr.lip6.fr/~chaillou/Public/DA-OCAML).

### Description
This is the original French version of the [Developing Applications With Objective Caml](https://caml.inria.fr/pub/docs/oreilly-book/) book.

### How do we know it's really free?

The author has made the book available on the author's website: https://www-apr.lip6.fr/~chaillou/

### For book lists, is it a book? For course lists, is it a course? etc.

This is a book.

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/master/CONTRIBUTING.md)
- [x] Search for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction)

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Comment by LuigiImVector** [comment]
To avoid creating another category with only one resource you could merge it with the "CAML" category (perhaps renaming it "CAML / OCAML") created in #6608

File: books/free-programming-books-fr.md [hunk 1]
```diff
@@ -11,7 +11,7 @@
 * [Assembleur](#assembleur)
 * [Bash / Shell](#bash--shell)
 * [C / C++](#c--c)
-* [Caml](#caml)
+* [Caml / OCaml](#caml)
```
Comment (LuigiImVector) [inline books/free-programming-books-fr.md:14]:
```suggestion
* [Caml / OCaml](#caml--ocaml)
```

The link didn't work

**Comment by LuigiImVector** [comment]
To avoid conflicts you could implement the changes made on #6608 in this PR

**Comment by davorpa** [comment]
Linter error goes from https://github.com/EbookFoundation/free-programming-books/pull/6426#issuecomment-968131507. Solved at #6611.

Please merge with upstream again

**Pull request merged**